lib/manager/npm/post-update/yarn.ts

```diff
@@ -94,7 +94,9 @@ export async function generateLockFile(
     // rangeStrategy = update-lockfile
     const lockUpdates = upgrades
       .filter((upgrade) => upgrade.isLockfileUpdate)
-      .map((upgrade) => upgrade.depName);
+      .map((upgrade) =>
+        [upgrade.depName, upgrade.currentValue].filter(Boolean).join('@')
+      );
```

Comment:
??? I think `currentValue` should never be `undefined`
```suggestion
      .map((upgrade) => `${upgrade.depName}@${upgrade.currentValue}`);
```

Does this work for older yarn versions too?

Reply:
:+1: will update.

Re older yarn versions, I checked on archive.org (!), and it's a documented way of using `yarn upgrade` for >3 years: https://web.archive.org/web/20170319085256/https://yarnpkg.com/lang/en/docs/cli/upgrade/